### PR TITLE
Update macOS troubleshooting for IOS build 

### PR DESCRIPTION
**Pull request by sealch**
Building React native app on macOS caused error : `/joplin/ReactNativeClient/ios/Pods/Target Support Files/Pods-Joplin/Pods-Joplin.debug.xcconfig: unable to open file (in target "Joplin" in project "Joplin") (in target 'Joplin' from project 'Joplin')` for me so I decided to update the troubleshooting section with the fix if someone else would be stuck.

### CODE REVIEW

**Comment by sealch** [comment]
@PackElend please label me

File: readme/build_troubleshooting.md [hunk 1]
```diff
@@ -21,6 +21,13 @@ If you get a node-gyp related error, you might need to manually install it: `npm
 
 If you get the error `libtool: unrecognized option '-static'`, follow the instructions [in this post](https://stackoverflow.com/a/38552393/561309) to use the correct libtool version.
 
+If you there is an error `/joplin/ReactNativeClient/ios/Pods/Target Support Files/Pods-Joplin/Pods-Joplin.debug.xcconfig: unable to open file (in target "Joplin" in project "Joplin") (in target 'Joplin' from project 'Joplin')` run following commands:
+
+    cd ios
+    pod deintegrate
+    pod install
+
+
```
Comment (laurent22) [inline readme/build_troubleshooting.md]: Please create a new section `# Mobile application`, then a sub-section `## iOS` and put your text there.

File: readme/build_troubleshooting.md [hunk 2]
```diff
@@ -36,4 +43,4 @@ This is an indication that there's an early initialisation error. Try this:
 
 **You should not use WSL at all** because this is a GUI app that lives outside of WSL, and the WSL layer can cause all kind of very hard to debug issues. It can also lock files in node_modules that cannot be unlocked when the app crashes. (You need to restart your computer.) Likewise, don't run the TypeScript watch command from WSL.
 
-So everything should be done from a Windows Command prompt or Windows PowerShell running as Administrator. All build and start commands are designed to work cross-platform, including on Windows.
\ No newline at end of file
+So everything should be done from a Windows Command prompt or Windows PowerShell running as Administrator. All build and start commands are designed to work cross-platform, including on Windows.
```
Comment (laurent22) [inline readme/build_troubleshooting.md]: Undo this change please.

File: readme/build_troubleshooting.md (hunk 1 above)
Comment (sealch) [reply readme/build_troubleshooting.md]: > Please create a new section `# Mobile application`, then a sub-section `## iOS` and put your text there.
Done. Please review my changes.

File: readme/build_troubleshooting.md (hunk 2 above)
Comment (sealch) [reply readme/build_troubleshooting.md]: > Undo this change please.

Done.

File: readme/build_troubleshooting.md (hunk 1 above)
Comment (laurent22) [reply readme/build_troubleshooting.md]: I wrote `# Mobile application` and `## iOS` and you've added `#Mobile application` and `##IOS`, which is not even valid Markdown. I wrote "iOS" and you wrote "IOS". Details are important in software development.

Comment (sealch) [reply readme/build_troubleshooting.md]: Sorry about that. WebStorm renders ##iOS without any issues, must be some mistake
